middleware/cache/cache.go

```diff
@@ -109,43 +109,47 @@ func New(config ...Config) fiber.Handler {
 		// Get timestamp
 		ts := atomic.LoadUint64(&timestamp)
 
-		// Check if entry is expired
-		if e.exp != 0 && ts >= e.exp {
-			deleteKey(key)
-			if cfg.MaxBytes > 0 {
-				_, size := heap.remove(e.heapidx)
-				storedBytes -= size
-			}
-		} else if e.exp != 0 {
-			// Separate body value to avoid msgp serialization
-			// We can store raw bytes with Storage 👍
-			if cfg.Storage != nil {
-				e.body = manager.getRaw(key + "_body")
-			}
-			// Set response headers from cache
-			c.Response().SetBodyRaw(e.body)
-			c.Response().SetStatusCode(e.status)
-			c.Response().Header.SetContentTypeBytes(e.ctype)
-			if len(e.cencoding) > 0 {
-				c.Response().Header.SetBytesV(fiber.HeaderContentEncoding, e.cencoding)
-			}
-			if e.headers != nil {
-				for k, v := range e.headers {
-					c.Response().Header.SetBytesV(k, v)
+		// Check if no-cache
+		if !cfg.noCache(c) {
```

Comment:
please flatten

simply turn over and make return, of course via the start of the mutex lock

Reply:
Do you mean to include these lines inside the check if no-cache condition?

// Lock entry
mux.Lock()

// Get timestamp
ts := atomic.LoadUint64(&timestamp)

Reply:
I've updated the file. Instead of enclosing the check/get entry, I put no-cache check condition on line 199 since it blocks the check if the entry is expired condition.

![image](https://user-images.githubusercontent.com/87841930/196570202-9302ce09-e9c5-4c81-a700-9f27e11a7426.png)

Reply:
i see no change in the structure ?